dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java

```diff
@@ -285,94 +285,89 @@ private void goToParentDirectory() {
     }
 
     private void transferToBlackPearl() {
-        try {
-            if (ds3Common.getCurrentSession() == null) {
-                alert.showAlert(resourceBundle.getString("noSession"));
-                return;
-            }
-            final ObservableList<TreeItem<FileTreeModel>> currentSelection = treeTable.getSelectionModel().getSelectedItems();
+        if (ds3Common.getCurrentSession() == null) {
+            alert.showAlert(resourceBundle.getString("noSession"));
+            return;
+        }
+        final ObservableList<TreeItem<FileTreeModel>> currentSelection = treeTable.getSelectionModel().getSelectedItems();
 
-            if (currentSelection.isEmpty()) {
-                alert.showAlert(resourceBundle.getString("fileSelect"));
-                return;
-            }
-            final TreeTableView<Ds3TreeTableValue> ds3TreeTableView = ds3Common.getDs3TreeTableView();
-            ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTableView.getSelectionModel().getSelectedItems()
-                    .stream().collect(GuavaCollectors.immutableList());
-            //Finding root in case of double click to get selected items
-            final TreeItem<Ds3TreeTableValue> root = ds3TreeTableView.getRoot();
-            if ((root == null || null == root.getValue()) && Guard.isNullOrEmpty(values)) {
-                alert.showAlert(resourceBundle.getString("selectDestination"));
-                return;
-            }
-            //If values is empty we have to assign it with root
-            else if (Guard.isNullOrEmpty(values)) {
-                final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
-                values = builder.add(root).build().asList();
-            }
-            if (values.size() > 1) {
-                alert.showAlert(resourceBundle.getString("multipleDestError"));
-                return;
-            }
+        if (currentSelection.isEmpty()) {
+            alert.showAlert(resourceBundle.getString("fileSelect"));
+            return;
+        }
+        final TreeTableView<Ds3TreeTableValue> ds3TreeTableView = ds3Common.getDs3TreeTableView();
+        ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTableView.getSelectionModel().getSelectedItems()
```

Comment:
Should this be `final`

Reply:
Same issue where root path is reassigned if there's no local selection